secure_message/authentication/jwe.py

```diff
@@ -95,12 +104,20 @@ def _load_keys(self, private_key, private_key_password, public_key):
 
         self.public_key = backend.load_pem_public_key(public_key_bytes)
 
+    def _decode_token(self, token):
+        """
+        Decodes string tokens without altering bytes tokens.
+        """
+        try:
+            return token.decode()
+        except AttributeError:
+            return token
+
     def decrypt_token(self, encrypted_token):
         """
         Function to decrypt encrypted jwt
         """
-        if not isinstance(encrypted_token, str):
-            encrypted_token = encrypted_token.decode()
+        encrypted_token = self._decode_token(encrypted_token)
```

Comment:
🎨 This line reads weirdly to me. Yes it's less efficient to create a new variable when we could recycle the old one, but after decoding the token then shouldn't it be 'decoded_token' or even just 'token'.  Not super important though

Reply:
I'm going to keep this as encrypted_token for now as the token is still fundamentally the same but is just going from a bytes object to a str.

Reply:
I agree took me a while to figured out what _decode_token was actually doing, I actually preferred it before. However all of this could go if we used the crypto library